src/components/biggive-carousel/biggive-carousel.tsx

```diff
@@ -0,0 +1,103 @@
+import { Component, Prop, Element, h } from '@stencil/core';
+import { brandColour } from '../../globals/brand-colour';
+import { spacingOption } from '../../globals/spacing-option';
+
+@Component({
+  tag: 'biggive-carousel',
+  styleUrl: 'biggive-carousel.scss',
+  shadow: true,
+})
+export class BiggiveCarousel {
+  @Element() host: HTMLBiggiveCarouselElement;
+
+  @Prop() spaceBelow: spacingOption = 4;
+
+  @Prop() columnCount: number = 3;
+
+  @Prop() buttonBackgroundColour: brandColour = 'white';
+
+  @Prop() buttonIconColour: brandColour = 'primary';
+
+  currentTab = 0;
+  itemCount = 0;
+  itemWidth = 0;
+  max = 0;
+  min = 0;
+  sleeve;
+
+  componentDidRender() {
+    this.sleeve = this.host.shadowRoot?.querySelector<HTMLElement>('.sleeve');
+    let children: Array<any> = Array.from(this.host.children);
+
+    this.itemCount = children.length;
+
+    if (children.length > 0 && this.sleeve) {
+      this.itemWidth = children[0].offsetWidth;
+      this.sleeve.style.width = this.itemWidth * children.length + 'px';
+
+      children.forEach(function (el) {
+        el.style.width = 'calc( 100% / ' + children.length + ' )';
+      });
+    }
+
+    this.max = 0;
+    this.min = 0 - (children.length - this.columnCount) * this.itemWidth;
+  }
+
+  showTab(i: number) {
```

Comment:
Maybe a docblock like this would help:

Not sure if it's better to do nothing if is out of range as you've got it, or throw an error.

```suggestion
  /**
    * Animates a transition to show the i'th element in the carousel, counting from zero. Does nothing if i out of range.
    */
  showTab(i: number) {
```